cfme/infrastructure/resource_pool.py

```diff
@@ -122,14 +161,15 @@ class ResourcePool(Pretty, Navigatable):
     """
     pretty_attrs = ['name', 'provider_key']
 
-    def __init__(self, name=None, provider_key=None, appliance=None):
-        Navigatable.__init__(self, appliance=appliance)
+    def __init__(self, name=None, provider_key=None, collection=None):
         self.quad_name = 'resource_pool'
         self.name = name
         if provider_key:
-            self.provider = get_crud(provider_key, appliance=appliance)
+            self.provider = get_crud(provider_key, appliance=self.collection.appliance)
         else:
             self.provider = None
+        self.collection = collection or ResourcePoolCollection()
```

Comment:
**REQUIRED**: This line should, at the very least, pass the appliance to ```ResourcePoolCollection()``` and the ```ResourcePoolCollection``` object should have an ```__init__``` which should set the appliance, it should not accept ```None```, an appliance should ALWAYS be passed to a collection.

Reply:
@psav but `ResourcePoolCollection` inherits `Navigatable` which has `__init__()` with appliance argument. So I think this will be enough:
```python
class ResourcePool(Pretty, Navigatable):
    ...
    def __init__(self, name=None, provider_key=None, collection=None, appliance=None):
        ...
        self.collection = collection or ResourcePoolCollection(appliance=appliance)
```